src/Azure.Functions.Cli/Helpers/AzureHelper.cs

```diff
@@ -449,11 +473,63 @@ private static async Task ArmHttpAsync(HttpMethod method, Uri uri, string access
                   }, 18, TimeSpan.FromSeconds(3)
             );
         }
+        public static async Task CreateFunctionAppOnContainerService(string accessToken, string managementURL, string subscriptionId, string resourceGroup, ContainerAppsFunctionPayload payload)
+        {
+            var url = new Uri($"{managementURL}/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Web/sites/{payload.Name}?api-version={ArmUriTemplates.FunctionAppOnContainerAppsApiVersion}");
+            ColoredConsole.WriteLine(Constants.FunctionAppDeploymentToContainerAppsMessage);
+            var response = await ArmClient.HttpInvoke(HttpMethod.Put, url, accessToken, payload);
+            if (!response.IsSuccessStatusCode)
+            {
+                try
+                {
+                    var content = response.Content.ReadAsStringAsync().Result;
+                    var errorResponse = JsonConvert.DeserializeObject<ContainerAppsFunctionDeployResponse>(content);
+                    throw new CliException($"The function app deployment to Container Apps failed. {(!string.IsNullOrWhiteSpace(errorResponse?.Message)? $"Error: {errorResponse.Message}" : string.Empty )}");
+                }
+                catch (CliException)
+                {
+                    throw;
```

Comment:
Why is this try-catch block necessary? It seems the only other places where an exception can occur here are when the content is being read and deserialized, at which point I think the specified methods would throw their own exceptions anyway.

Reply:
I don't understand the statement. ` at which point I think the specified methods would throw their own exceptions anyway.`. 

Yes, the idea is the same. The backend can't guarantee that content will always be JSON when the result is not `success`. Therefore, `JsonConvert.DeserializeObject` can throw exception. We need to catch that. The idea is that if `Message` field cannot be read from the content for any reason/exception then the default message is delivered in the UI. 

I am simplifying the try/catch through.

Reply:
Right, my concern was primarily about rethrowing the same exception since that would obfuscate the stack trace. I think in general, unless we want to throw our own custom `CliException` that it would be best for the exception to originate from the line in which the code failed rather than rethrowing. However, this appears to be addressed with your recent changes. Thanks!